src/main/java/uk/gov/hmcts/sptribs/caseworker/service/HearingService.java

```diff
@@ -114,30 +119,37 @@ public void addListing(CaseData caseData, Listing listing) {
         }
     }
 
+    @SneakyThrows
     public void updateHearingList(CaseData caseData) {
+        String hearingName = caseData.getCicCase().getHearingList().getValue().getLabel();
         for (ListValue<Listing> listingListValue : caseData.getHearingList()) {
-            String hearingName = caseData.getCicCase().getHearingList().getValue().getLabel();
             if (isMatchingHearing(listingListValue, hearingName)) {
-                listingListValue.setValue(caseData.getListing());
+                // deep copy of listing needed in order to clear rec file
+                Listing listingDeepCopy = objectMapper
+                    .readValue(objectMapper.writeValueAsString(caseData.getListing()), Listing.class);
+                listingListValue.setValue(listingDeepCopy);
                 break;
             }
         }
     }
 
+    @SneakyThrows
     public void updateHearingSummaryList(CaseData caseData) {
+        String hearingName = caseData.getCicCase().getHearingSummaryList().getValue().getLabel();
```

Comment:
As the difference between this method (updateHearingSummaryList) and updateHearingList seems only to be in the retrieval of getHearingSummaryList() instead of getHearingList(), is it worth having one method (instead of two) which takes in two parameters, the caseData and some Boolean to determine whether the HearingList or the HearingSummaryList is updated? Or would that not make much difference or confuse things?